src/server/api/contactsApi.js

```diff
@@ -0,0 +1,165 @@
+import { r } from '../models'
+import _ from 'lodash'
+import { getTimezoneByZip, getOptOutSubQuery } from '../../workers/jobs'
+
+import { getValidatedData } from '../../lib'
+
+function campaignaStatusShortCircuit(campaign, res) {
+  let message = ''
+  if (campaign.is_archivedj) {
+    message = 'Campaign is archived'
+  } else if (campaign.is_started) {
+    message = 'Campaign is started'
+  }
+
+  if (message) {
+    res.writeHead(403)
+    res.end(message)
+    return true
+  }
+
+  return false
+}
+
+// TODO(lperson) enforce that this is not reentrant
+// TODO(lperson) enforce auth
+export default async function contactsApi(req, res) {
+  const orgId = req.params.orgId
+  const campaignId = req.params.campaignId
+
+  if (!['GET', 'DELETE', 'POST'].includes(req.method)) {
+    res.writeHead(405, { Allow: 'GET, POST, DELETE' })
+    res.end('Not allowed')
+    return
+  }
+
+  const campaignArray = await r
```

Comment:
I really like the pattern `const [campaign] = await r.knex(...)... ; if (!campaign) ...` but this is fine.

Reply:
👍